practical/DeepClustering/FlorianKittel/kmeans.py

```diff
@@ -0,0 +1,48 @@
+import torch
+
+
+class MiniBatchKMeans:
+    def __init__(self, k: int, batch_size: int, iterations: int):
+
+        if k <= 0:
+            raise ValueError("k must be greater than 0")
+
+        if batch_size <= 0:
+            raise ValueError("batch_size must be greater than 0")
+
+        if iterations < 0:
+            raise ValueError("iterations must be greater than or equal to 0")
+
+        self.k = k
+        self.batch_size = batch_size
+        self.iterations = iterations
+
+    def __calculate_distance__(self, a: torch.Tensor, b: torch.Tensor) -> torch.Tensor:
+        return torch.sqrt(torch.sum((a - b) ** 2))
+
+    def __get_random_indices__(self, k: int, X: torch.Tensor) -> torch.Tensor:
+        return torch.randperm(len(X))[:k]
```

Comment:
Use either `torch.randint` or `numpy.random.choice` to properly select your random indexes. The permutation result will be very large for large datasets, resulting in higher memory usage.